packages/iov-tendermint-rpc/src/encodings.ts

```diff
@@ -156,8 +157,13 @@ export class Base64 {
 }
 
 export class DateTime {
-  public static decode(dateTimeString: DateTimeString): ReadonlyDate {
-    return Encoding.fromRfc3339(dateTimeString);
+  public static decode(dateTimeString: DateTimeString): ReadonlyDateWithNanoseconds {
+    const readonlyDate = Encoding.fromRfc3339(dateTimeString);
+    const nanosecondsMatch = dateTimeString.match(/\.(\d+)Z$/);
+    const nanoseconds = nanosecondsMatch ? nanosecondsMatch[1] : "";
+    // tslint:disable-next-line:no-object-mutation
+    (readonlyDate as any).nanoseconds = parseInt(nanoseconds.padEnd(9, "0"), 10);
+    return readonlyDate as ReadonlyDateWithNanoseconds;
```

Comment:
If JavaScript's _everything is an object mentality_ is true, then this code should do the same:

```ts
    const readonlyDate = Encoding.fromRfc3339(dateTimeString);
    const nanosecondsMatch = dateTimeString.match(/\.(\d+)Z$/);
    const nanoseconds = nanosecondsMatch ? nanosecondsMatch[1] : "";
    return {
      ...readonlyDate,
      nanoseconds: parseInt(nanoseconds.padEnd(9, "0"), 10),
    };
```

At least TypeScript is happy. Not runtime tested.